flang/test/Intrinsics/adjust.f90

```diff
@@ -0,0 +1,12 @@
+program test_adjust
```

Comment:
You just added this file in your latest commit, was it on purpose ?
We would definitely benefit from some simple end-to-end tests, but right now we are not quite sure how best to add that while respecting llvm modular testing principles.

Reply:
Yes and no!  :-)

I used it for my own end-to-end test and wanted to keep it around to start pondering how to integrate it the bigger picture testing.  Then I accidentally added it to the commit.   I do think we should start adding these tests (it was certainly valuable for my testing -- and nice to see it all work correctly) but I'm also at a loss re: rolling it into the overall testing infrastructure. I'd prefer to have a home for such tests than wait -- I guess I'm now (accidentally ;-) pushing for more discussions on this front.

Reply:
Well, the sooner we agree on a strategy here the better. I fully agree it's best to write and keep track of those tests while implementing lowering for them.

What I am wondering is what format we should use for them (how the output will be automatically checked ...). We have a few end-to-end tests using lit in fir-dev (e.g. https://github.com/flang-compiler/f18-llvm-project/blob/fir-dev/flang/test/Examples/hello.f90), but I am not sure this is the way the end-to-end tests should be written in the end.

I imagine we would want them in a format so that they can placed and run in: https://github.com/llvm/llvm-test-suite/tree/main/Fortran/UnitTests

Maybe we should ping naromero77 for inputs here on how that will work, and if we should start adding those directly in the llvm-test-suite rather than here. Right now there is no official tools we could pass as `CMAKE_Fortran_COMPILER` to test fir-dev. We would want to be sure that these tests will work with both fir-dev and other mainstream fortran compilers.

I am not against starting stacking those here while we are having the discussion, but then let's go all the way and directly add a *.reference_output for it.

Reply:
I don't like our chances of upstreaming end to end tests included in the "regressions" (a.k.a. flang/test) area. On the other hand, I don't think we've reached any agreement on where flang will put its end to end Fortran tests either.

Reply:
Agreed -- for now it seemed like a good spot for the end-to-end intrinsics tests to land but feel free to move as you see fit.